cli/lockfile.rs

```diff
@@ -114,6 +144,33 @@ impl Lockfile {
     }
   }
 
+  pub fn check_or_insert_npm_package(
+    &mut self,
+    package: &NpmResolutionPackage,
+  ) -> bool {
+    if self.write {
+      // In case --lock-write is specified check always passes
+      self.insert_npm_package(package);
+      true
+    } else {
+      self.check_npm_package(package)
+    }
+  }
+
+  pub fn check_or_insert_npm_specifier(
+    &mut self,
+    package_req: &NpmPackageReq,
+    version: String,
+  ) -> bool {
```

Comment:
Returning just a `bool` here is not so great - ideally we should return `Result<(), AnyError>` (or more specialized error type) that could provide context info what went wrong. That will require changes in `deno_graph`

Reply:
It looks like we could at least change code like check_npm_package to return a result.